src/app/@ansyn/status-bar/components/tree-view/tree-view.component.ts

```diff
@@ -79,34 +108,42 @@ export class TreeViewComponent implements OnInit, OnDestroy {
 		);
 	}
 
-	updateInputDataFilterMenu(): void {
-		if (Boolean(this._selectedFilters)) {
-			this.dataInputFiltersItems.forEach((dataInputItem) => {
-				// first iterate the parents and update the their checkboxes.
-				dataInputItem.children.forEach((sensor) => {
-					sensor.checked = this._selectedFilters.some(selectedFilter => isEqual(selectedFilter, sensor.value));
-				});
-				// then iterate all the children and update the their checkboxes.
-				if (dataInputItem.children.some(child => child.checked)) {
-					// true = All / false = None / undefined = Partial
-					dataInputItem.checked = dataInputItem.children.every(child => child.checked) ? true : undefined;
-				}
-				else {
-					dataInputItem.checked = false;
-				}
-			});
+	updateInputDataFilterMenu(curr: TreeviewItem): void {
+		if (!Boolean(this._selectedFilters)) {
+			return;
+		}
+
+		if (this.isLeaf(curr)) {
+			curr.checked = this._selectedFilters.some(selectedFilter => isEqual(selectedFilter, curr.value));
+			return;
 		}
+		curr.children.forEach(c => this.updateInputDataFilterMenu(c));
+		curr.checked = this.treeViewNodeStatus(curr);
 	}
 
+	isLeaf(node: TreeviewItem) {
+		return !(Array.isArray(node.children) && node.children.length > 0);
+	}
+
+	treeViewNodeStatus(node: TreeviewItem): boolean {
+		return node.children.every(child => child.checked) ? true : node.children.some(child => child.checked || child.checked === undefined) ? undefined : false;
+	}
 
 	dataInputFiltersOk(): void {
-		this.store.dispatch(new SetOverlaysCriteriaAction({
-			dataInputFilters: {
-				filters: this._selectedFilters,
-				active: this.dataInputFiltersActive
-			}
-		}));
-		this.closeTreeView.emit();
+		if (this.dataInputFiltersItems.some((provider) => provider.checked || provider.checked === undefined) || !this.dataInputFiltersActive) {
+			this.store.dispatch(new SetOverlaysCriteriaAction({
+				dataInputFilters: {
+					fullyChecked: this.leavesCount <= this._selectedFilters.length,
+					filters: this._selectedFilters,
+					active: this.dataInputFiltersActive
+				}
+			}));
+			this.closeTreeView.emit();
+		} else {
+			this.store.dispatch(new SetToastMessageAction({
```

Comment:
I thought we should support a scenario where all sensors are unchecked. I guess it is unnecessary only in case we fill the configuration with the full sensors list.